janitor/functions.py

```diff
@@ -2771,7 +2771,18 @@ def select_columns(
         This will result in selection of the complement of the columns
         provided.
     :returns: A pandas DataFrame with the specified columns selected.
+    :raises: NameError if one or more of the specified column names are
+        not found in DataFrame columns.
     """
+    wildcards = {col for col in search_column_names if "*" in col}
```

Comment:
Thanks for the feedback @hectormz! Yes, since this PR suggests raising a `NameError` for "standard" column names, it seems reasonable to also raise the same error for missing search strings. I have implemented a suggestion for review.

Reply:
great, thanks @smu095